src/Schema/Tables/Contracts/Table.php

```diff
@@ -220,6 +220,28 @@ public function empty_table() {
 
 		$this_table = static::table_name( true );
 
+		$this->db::query( "SET foreign_key_checks = 0" );
+		$result = $this->db::query( "DELETE FROM {$this_table}" );
+		$this->db::query( "SET foreign_key_checks = 1" );
+
+		return $result;
+	}
+
+	/**
+	 * Truncates the custom table.
+	 *
+	 * @since 1.1.8
+	 *
+	 * @return int|false The number of removed rows, or `false` to indicate a failure.
+	 */
+	public function truncate() {
```

Comment:
If this is going to be a public method, shouldn't this be added to the [Schema_Interface](https://github.com/stellarwp/schema/blob/ed6f8e1bebcff3bb21b2407298876672cf0ff362/src/Schema/Tables/Contracts/Schema_Interface.php) and a major version release (e.g this is a breaking change) be released?

Reply:
@defunctl i am not against that - i just didnt consider it necessary. Happy to follow what you think is best in this case.